data/monitoring/prom-rule.yaml

```diff
@@ -11,6 +11,20 @@ spec:
       rules:
         - expr: sum(up{namespace='{{ .Namespace }}', pod=~'cluster-network-addons-operator-.*'} or vector(0))
           record: kubevirt_cnao_num_up_operators
+        - expr: sum(kube_deployment_labels{deployment=~'kubernetes-nmstate-operator'} or vector(0))
+          record: kubevirt_nmstate_operator_deployments
+        - expr: sum(kube_daemonset_labels{namespace='{{ .Namespace }}', daemonset=~'nmstate-handler'} or vector(0))
+          record: kubevirt_cnao_nmstate_deployed
+        - alert: CnaoNmstateMigration
+          annotations:
+            summary: Nmstate will be removed from CNAO.
+            runbook_url: http://kubevirt.io/monitoring/runbooks/CnaoNmstateMigration
+          expr: sum(kubevirt_cnao_nmstate_deployed or vector(0)) > 0 and sum(kubevirt_nmstate_operator_deployments or vector(0)) == 0
+          for: 5m
+          labels:
+            severity: warning
+            kubernetes_operator_part_of: kubevirt
+            kubernetes_operator_component: cluster-network-addons-operator
```

Comment:
we want to highlight this alert in the UI virtualization overview page.

can we add a label that will tell the UI to show this alert on the top of the virtualization overview page for better visibility ?

for exampe:
`kubevirt_ui: highlighted`
does that makes sense?

Reply:
do we have a naming convention for alert labels?

Reply:
I don't think we have, but @sradco may confirm that.

Reply:
I would suggest

```
kubevirt_console_scope: [cluster,virtualmachine]
```
But IIUC, we need to provide more info in case an alert is cluster scoped (that's not case of the alert in this PR, just to make the mechanism general).

Reply:
@yaacov Can you please explain what is the reasoning to highlight this specific alert? 
I believe alerts should be ordered based on their severity.

Reply:
`kubevirt_console_scope: [cluster,virtualmachine]`
is useful to specify the scope,
we will also need a way to specify that we want this alert to be shown in the top of the overview page.
`kubevirt_console_popup: [cluster,virtualmachine]`
?

Reply:
@sradco We want to show this specific alert in OCP Console UI as a banner on CNV overview page

Reply:
@yaacov My thinking was that is the kubevirt_console_scope is define, then we can implicitly assume that we want to show that alert in UI, no? Otherwise, if it's not there, UI wouldn't show it

Reply:
@yaacov please see my question above. I don;t understand the need for this.
Having these "console" details in the alert as labels will be wrong practice. 
If you want to do this, it should be in the UI level only.
The alerts can be used is multiple envs, not only OCP UI and are also sent to the user with alert manager with all its labels. 
Adding this will create clutter that is not relevant to the alert itself.
The new labels we added about the operator name and component are directly related to the alert details and help the users to understand where the issue comes from.

Reply:
@sradco hi,

> the reasoning to highlight this specific alert?

IIUC the request is that an alert be shown on OCP Console UI as a banner on CNV overview page.
UI need some hint that this alert request to be shown this way.
( all alerts including this one will be shown in the regular alerts list as well. )

> If you want to do this, it should be in the UI level only.

AFAIU this is a property of the alert, the UI read this hint, and decide to show alerts that has this property differently.
this alert can then be updated by the people who create the alert rule, and decide to not show it if not needed anymore, or create a different alert that request to be displayed on top of the page.

It's an independent property from the severity, for example an alert can by warning or information and request to be highlighted.

Reply:
@sradco
do we currently have a way to know that an alert belong to a specific virtual machine?

AFAIU the request was that this alert will be shown in the virtualization overview, and another for a specific vm that needs some action because of some event.

Reply:
I still think this should be done on the UI level. 
Perhaps there could be a place to state the alerts to show in the banner that is separate from the alerts . Then the UI can check based on it.
I strongly believe we should not  add these labels to the alerts.

Reply:
We don't have alerts for specific VMs at this point. There are planned alerts for 4.12.

Reply:
@sradco We need a way to surface specific alerts so they are not "lost" in all the other alerts.
Those alert may have different severity, like a reboot is required, which can be warning, but the user must be aware of it.
The the current case, users that are using NMstate from CNV must install the stand alone operator before moving to 4.11. In this case it's not a critical alert the requires immediate action, but it is an important one. Having it as a warning will probably not be visible enough.

Reply:
@sradco We're looking for a generic way of displaying an alert in UI (not in Observe/metrics).
Marking an alert with an annoration/label/something allows leaving all logic for the backend, and UI only presents
what it's supposed to.
Wiring logic into UI, what is supposed to be displayed, what not, is very sub-optimal.

> I strongly believe we should not add these labels to the alerts.
Could you explain please?

I believe having an extra annotation would be a huge clutter for the users.
At the very least, the gain we get - a user being notified on a relevant place in the UI, instead of somewhere in Monitoring category, seems to be prevalent to me

Reply:
> Perhaps there could be a place to state the alerts to show in the banner that is separate from the alerts.

Also works for me, my concern is that maintainers that will update the alert rules forget to update the alert rules "impact to user" table that will be in a different file.
It will also be easier to a user looking at the active alert to filter by "impact to user", in case they don't use the UI.

Reply:
In that case I think we should not use UI in the label name.
It should be somehow reflected that an action is required or something like that.
Like a sub label to severity.

Reply:
> It should be somehow reflected that an action is required

`admin_action_recommended: <some longer description>` 
or 
`recommended_action: yes/[no]` 

does that makes more sense?

Reply:
I thought of something like:
severity_score / severity_level / severity_priority with  a few optional values, but not sure what they should be called.

Reply:
> Like a sub label to severity.

`severity_priority` makes sense to me because it indicate that handling this alert requires higher priority even if it's currently only a warning.

> What will happen if there are multiple ones?

They will all show in the top of the page, if they all require user to be aware of them, without raising them to critical and without hiding theme between all the other warnings and information alerts.

Reply:
maybe event just `priority` , without the `sevirity_`  prefix ?

Reply:
we will also need to document how to use the labels, we don't want every critical alert to also be high priority (because it's critical)

Reply:
I found a nice article that supports the naming and values for the label:
https://www.guru99.com/defect-severity-in-software-testing.html

I think we should keep it as priority.

Reply:
The documentation for the label should be part of OCP metrics documentation.
Im not sure if we should have it as mandatory as the severity. 
The issue is that the label naming impacts also OCP and all other operators.
We should propose this in the monitoring forum and get their opinion.

Reply:
> I think we should keep it as priority.

makes sense to me too

> Im not sure if we should have it as mandatory as the severity.

IMHO it should not be mandatory (default to medium if missing) 
my reasoning is that unlike severity that is clear to define by the rule maintainer, priority may depend on things outside the scope of the alert rules maintainer, the cases where the rule maintainer wish to raise/lower priority are the odd cases.

Reply:
I created https://github.com/openshift/enhancements/pull/1077. Please review.

Reply:
from https://github.com/openshift/enhancements/pull/1077 is looks like "priority: high"

@rhrazdil can you add the "priority" label?

Reply:
Sorry, I've missed your comment.
Added the `priority: high` label in a new commit